script/interface.php

```diff
@@ -15,8 +15,35 @@
 function _get(&$db, $case) {
 	switch ($case) {
 		case 'tasks' :
-			
-			print json_encode(_tasks($db, (int)$_REQUEST['id_project'], $_REQUEST['status'], GETPOST('fk_user'), GETPOST('fk_soc'), GETPOST('soc_type') ));
+			$task = new Task($db);
+			$extrafieldstask = new ExtraFields($db);
+			$extrafieldstask->fetch_name_optionals_label($task->table_element);
+			$search_array_options = $extrafieldstask->getOptionalsFromPost($task->table_element, '', 'search_');
+			$dates = array(
+                'start_date_after',
+                'start_date_before',
+                'end_date_after',
+                'end_date_before'
+            );
+			$TDateFilters = array_map(
+			    function ($datePrefix) use ($db) {
+			        $h = $m = $s = 0;
+			        if (preg_match('/.*before$/', $datePrefix)) {
+			            $h = 23;
+			            $m = $s = 59;
+                    }
+			        return $db->idate(
+			            dol_mktime(
+			                $h, $m, $s,
+                            GETPOST($datePrefix . 'month'),
+                            GETPOST($datePrefix . 'day'),
+                            GETPOST($datePrefix . 'year')
+                        )
+                    );
+                },
+                $dates
+            );
+			print json_encode(_tasks($db, (int)$_REQUEST['id_project'], $_REQUEST['status'], GETPOST('fk_user'), GETPOST('fk_soc'), GETPOST('soc_type'), $TDateFilters, $search_array_options, $task, $extrafieldstask ));
```

Comment:
not pour plus tard, si c'est pas absolument nécessaire on évite d'utiliser $_REQUEST['xxxxxx'] on privilégie GETPOST('xxxxxx').
Là c'est bien sur pas toi, mais quand on en voit il faut se poser la question et changer SI nécessaire. (et seulement si nécessaire) :)

Reply:
J’ai fait le remplacement du coup.